libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/TreeViewRenderer/TreeViewRenderer.tsx

```diff
@@ -557,20 +501,94 @@ export class TreeViewRenderer extends React.PureComponent<
     this.setState(this.resizeFunc);
   }
 
-  private reloadData(treeNodes: any[]): void {
+  private reloadData(requestTreeNodes: IRequestNode[]): void {
     const reloadDataFunc = (
       state: Readonly<ITreeViewRendererState>
     ): ITreeViewRendererState => {
-      if (!treeNodes || treeNodes.length === 0 || !treeNodes[0]) {
+      if (
+        !requestTreeNodes ||
+        requestTreeNodes.length === 0 ||
+        !requestTreeNodes[0]
+      ) {
         return state;
       }
-      const rootSize = treeNodes[0].size;
-      const rootErrorSize = treeNodes[0].error;
+
+      const rootSize = requestTreeNodes[0].size;
+      const rootErrorSize = requestTreeNodes[0].error;
       const rootLocalError = rootErrorSize / rootSize;
 
+      const min: number = rootErrorSize / rootSize;
+      const max: number = Math.max(
+        ...requestTreeNodes.map((node) => {
+          if (node.size === 0) {
+            return 0;
+          }
+          return node.error / node.size;
+        })
+      );
+
+      const minColor = ColorPalette.MinColor;
+      const maxColor = ColorPalette.MaxColor;
+
+      const colorgrad = d3scaleLinear<string>()
+        .domain([min, max])
+        .interpolate(d3interpolateHcl)
+        .range([minColor, maxColor]);
+
+      // From the retrieved request, calculate additional properties
+      // that won't change during UI updates
+      const treeNodes = requestTreeNodes.map(
+        (node): ITreeNode => {
+          const globalErrorPerc = node.error / rootErrorSize;
+          const localErrorPerc = node.error / node.size;
+          const calcMaskShift = globalErrorPerc * 52;
+          const filterProps = this.calculateFilterProps(node, rootErrorSize);
+
+          let heatmapStyle: { fill: string } = { fill: errorAvgColor };
+
+          if (node.error / node.size > rootLocalError * errorRatioThreshold) {
+            heatmapStyle = { fill: colorgrad(localErrorPerc)! };
+          }
+
+          return {
+            arg: node.arg,
+            condition: node.condition,
+            error: node.error,
+            errorColor: heatmapStyle,
+            fillstyleDown: {
+              transform: `translate(0px, -${calcMaskShift}px)`
+            },
+            fillstyleUp: {
+              fill: "#d2d2d2",
```

Comment:
Do we want to put such hard-coded colors into a separate file? There are other files, too, that have this.

Reply:
done!